src/c2goto/library/io.c

```diff
@@ -37,6 +37,16 @@ __ESBMC_HIDE:;
   return ret;
 }
 
+char *gets(char *buf)
+{
+__ESBMC_HIDE:;
+  size_t i = 0;
+  size_t size = nondet_uint();
+  for(i = 0; i < size; i++)
+    buf[i] = nondet_char();
```

Comment:
From the documentation: "Reads [...] and stores [...] until a newline character or the end-of-file is reached." We should ensure these invariants.

Reply:
Fixed.